src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3844,6 +3844,11 @@ public override BoundNode VisitAsOperator(BoundAsOperator node)
 
         public override BoundNode VisitSuppressNullableWarningExpression(BoundSuppressNullableWarningExpression node)
         {
+            if (this._member.NonNullTypes == null)
+            {
+                Diagnostics.Add(ErrorCode.WRN_MissingNonNullTypesContext, node.Syntax.GetLocation());
```

Comment:
>ErrorCode.WRN_MissingNonNullTypesContext [](start = 32, length = 40)

Should this be reported in binding instead? We'll also want to report this in non-executable code: `static void F(object o = null!) { }`

Reply:
The reason I put it here rather than in binding was to avoid pulling on `NonNullTypes` early.
I'll add a PROTOTYPE to consider after the PR for warning on `string?` in wrong context is merged.

In non-executable code, there should already be an error so I'm not worried. Aren't we going to run `NullableWalker` on such code anyways?

---
In reply to: [205838382](https://github.com/dotnet/roslyn/pull/28825#discussion_r205838382) [](ancestors = 205838382)